src/Microsoft.AspNetCore.ResponseCaching/CacheEntry/CachedVaryRules.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    internal class CachedVaryRules
+    {
+        internal FastGuid Guid { get; set; }
```

Comment:
KeyPrefix. This could even be a string, at this point you don't care that it was derived from a FastGuid.